### PR TITLE
Add 'soil' package

### PR DESCRIPTION
**Pull request by frequem**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/create/cmake.html)
  step by step carefully. **[Yes]**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/frequem/hunter/builds/21244975
  * https://travis-ci.com/frequem/hunter/builds/95780803

* I have submitted CI configs to https://github.com/ingenue/hunter targeting `pkg.template` branch,
  see this merged pull request https://github.com/ingenue/hunter/pull/323

---

* I've checked this [Git style guide](https://0.readthedocs.io/en/latest/git.html). **[Yes]**
* I've checked this [CMake style guide](https://0.readthedocs.io/en/latest/cmake.html). **[Yes]**
* My change will work with CMake 3.2 (minimum requirement for Hunter). **[Yes]**
* I will try to keep this pull request as small as possible and will try not to mix unrelated features. **[Yes]**

---

### CODE REVIEW

File: docs/packages/pkg/soil.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+.. index:: 
+  single: graphics ; soil
+
+.. _pkg.soil:
+
+soil
+====
+
+-  https://www.lonesock.net/soil.html
+-  `Official <https://github.com/frequem/Simple-OpenGL-Image-Library>`__
```
Comment (ruslo) [inline docs/packages/pkg/soil.rst]: Doesn't look like an official repository

File: cmake/projects/soil/hunter.cmake [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    soil
+    VERSION
+    1.0.4
+    URL
+    "https://github.com/frequem/Simple-OpenGL-Image-Library/archive/v1.0.4.tar.gz"
```
Comment (ruslo) [inline cmake/projects/soil/hunter.cmake]: Do you want me to create https://github.com/hunter-packages/Simple-OpenGL-Image-Library instead? See https://docs.hunter.sh/en/latest/creating-new/patch.html for details.

File: docs/packages/pkg/soil.rst [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+-  `Example <https://github.com/frequem/hunter/blob/master/examples/soil/CMakeLists.txt>`__
```
Comment (ruslo) [inline docs/packages/pkg/soil.rst]: Since you're targeting https://github.com/ruslo/hunter it should be `ruslo/hunter` instead of `frequem/hunter`.

**Comment by Movebytes** [comment]
Any updates here?

File: docs/packages/pkg/soil.rst (hunk 1 above)
Comment (Movebytes) [reply docs/packages/pkg/soil.rst]: Official repository doesn't exist. SOIL library was copied and posted to github by @kbranigan from the http://lonesock.net/soil.htm. So 'official' there is only the web page.

Comment (frequem) [reply docs/packages/pkg/soil.rst]: I changed it to be a bit more clear and added kbranigan's version as 'GitHub' and mine as 'Hunterized'

File: cmake/projects/soil/hunter.cmake (hunk 2 above)
Comment (frequem) [reply cmake/projects/soil/hunter.cmake]: Not really necessary since it's not explicitly a hunter version and can also be used in standalone cmake.

**Comment by Movebytes** [comment]
@ruslo Please check it.

**Comment by ruslo** [comment]
Testing:
- https://ci.appveyor.com/project/ingenue/hunter/builds/21988636
- https://travis-ci.org/ingenue/hunter/builds/486243762?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.93

**Pull request merged**